srecord/input/catenate.cc

```diff
@@ -27,9 +28,9 @@ srecord::input_catenate::~input_catenate()
 }
 
 
-srecord::input_catenate::input_catenate(const pointer &a1, const pointer &a2) :
-    in1(a1),
-    in2(a2)
+srecord::input_catenate::input_catenate(pointer a1, pointer a2) :
+    in1(std::move(a1)),
```

Comment:
That's where I cannot really say if it's good or not. As developer of safety critical systems, where dynamic memory is forbidden, I never used this command.
If I understand the the description correctly, they would do this for objects of any size, because the only purpose of the const-reference was to copy the content once.

Reply:
I work on medical devices with similar restrictions. Bottom line is that I don't see the benefit we get in exchange for complexity in the source.

Reply:
I'm doing automotive, with less than 256KB of RAM. So I use native pointers.
That's why I do not have any real opinion on this. I'm ok to not apply the fix.

As I try to improve my knowledge on smart pointers, I had a deeper look into the code:
The object we're talking about is a pointer. In case of a native pointer I would definitely have done it by value. But smart pointers do have an overhead, so `const&` seems to be used everywhere to avoid that.
The fix only inserted a move, where a member variable is initialized (constructor). All other functions in the call stack were untouched. If this is on purpose, then the code gets definitely more complex: by value for variable init, but not for calling another function.

Reply:
I think performance isn't an issue here and I'd rather skip the complexity in the source. Let's leave it out and not apply this fix.